### PR TITLE
Factor out zingoconfig crate

### PR DESCRIPTION
**Pull request by zancas**
Within `wallet` all references to lightclient are via `lightclient_config`.

This defines a separate `zingoconfig` crate.  This has several benefits:
  * decouples `wallet` and `client` (not completely)
  * moves away from (accidental) circular dependencies
  * moves towards building `wallet` as a separate (WASM-able) concern

### CODE REVIEW

**Review by AloeareV — CHANGES_REQUESTED** [review]
If you check the ci, `cargo fmt check` failed. Can you push a commit with a `cargo fmt` run?

File: rustfmt.toml [hunk 1]
```diff
@@ -1,2 +1 @@
 edition = "2021"
-max_width = 120
```
[the remark this answers is not in the document]
Comment (AloeareV) [reply rustfmt.toml:2]: I do, yes.

**Review by AloeareV — APPROVED** [review]
Looks great!

File: rustfmt.toml [hunk 2]
```diff
@@ -1,2 +1,2 @@
...
\ No newline at end of file
+max_width = 100
```
Comment (zancas) [inline rustfmt.toml:2]: @AloeareV I believe this was your recommendation.

**Pull request merged**